floss/language/identify.py

```diff
@@ -89,22 +90,32 @@ def get_if_go_and_version(pe: pefile.PE) -> Tuple[bool, str]:
         b"\xfa\xff\xff\xff\x00\x00",
         b"\xf1\xff\xff\xff\x00\x00",
     ]
-
+    go_functions = [
+        b"runtime.main",
+        b"main.main",
+        b"runtime.gcWork",
+        b"runtime.morestack",
+        b"runtime.morestack_noctxt",
+        b"runtime.newproc",
+        b"runtime.gcWriteBarrier",
+        b"runtime.Gosched",
+    ]
     # look for the .rdata section first
-    for section in pe.sections:
-        try:
-            section_name = section.Name.partition(b"\x00")[0].decode("utf-8")
-        except UnicodeDecodeError:
-            continue
-        if ".rdata" == section_name:
-            section_va = section.VirtualAddress
-            section_size = section.SizeOfRawData
-            section_data = section.get_data(section_va, section_size)
-            for magic in go_magic:
-                if magic in section_data:
-                    pclntab_va = section_data.index(magic) + section_va
-                    if verify_pclntab(section, pclntab_va):
-                        return True, get_go_version(magic)
+    try:
+        section = get_rdata_section(pe)
+    except ValueError:
+        logger.debug(".rdata section not found")
+
+    section_va = section.VirtualAddress
```

Comment:
this now needs to be in an else clause so we don't end up here even if there's no .rdata section